src/artifact/v3/version/version.hpp

```diff
@@ -50,9 +53,16 @@ extern const ErrorCategoryClass ErrorCategory;
 
 error::Error MakeError(ErrorCode code, const string &msg);
 
-struct Version {
-	const int version;
-	const string format;
+class Version {
+public:
+	int version;
+	string format;
+
+	sha::SHA GetShaSum() const {
+		return shasum_;
+	}
```

Comment:
Since this was originally a struct, it has no constructor, and all members are still public, I would just keep the struct, remove this and use the member directly. The `_` suffix is reserved for private members to make the distinction between struct and class more prominent when reading, so that should also be removed if going down this route.

Reply:
I used [this other class](https://github.com/mendersoftware/mender/blob/121dbbd87c65344679c287b3f1e249a1372f3c4f/src/artifact/v3/manifest/manifest.hpp#L39) as a reference, which that the exact thing... But to be honest I agree with you that it breaks the convention and we should amend the other class instead. I'll fix

Reply:
Same here, it's just a value, not something that can/has to be calculated on demand.